rc/base/new-client.kak

```diff
@@ -0,0 +1,16 @@
+define-command new -params .. -command-completion -docstring '
+new [<commands>]: create a new kakoune client
+The ''terminal'' alias is being used to determine the user''s preferred terminal emulator
+The optional arguments are passed as commands to the new client' \
+%{
+    try %{
+        terminal %sh{
```

Comment:
We should not need to have a shell block here, in the end I think the terminal command should take a list of arguments to make forwarding cleaner.

Reply:
I don't think we can really make forwarding "clean" in this case, the `terminal` commands go through `sh -c` so the best they can do is join the arguments with a space. Just taking one argument makes things more explicit. 
As for the `new` command, we could bypass the shell block by doing `terminal "kak -e '%arg{@}'` but this has issues if the arguments to `new` contain single quotes. The current way seems better to me, even if it goes through a shell scope (which is negligible in this context).

Reply:
* I'd like `terminal` to use -shell-completion, so that I can just do `:terminal <use shell command completion here directly from kakoune>`, if terminal takes a single string parameter that will not be possible.
* `:new my command as separate args` should forward to `terminal kak -c %val{session} -e "%arg{@}"` which will pass "my commands as separate args" as a single argument to the `terminal` command, this is effectively not fool proof, but thats already the behaviour we have right now with the `:new` command. (try `:new echo %{foo     bar}`, you'll see it output just `foo bar`).
* The shell escaping should be an implementation detail of the `terminal` command implementation, as it depends on how that works, if the `terminal` command knows the argument is passed as-is, there is no need for escaping, on the contrary if for some reasons it has to foward it through `sh -c`, then it needs to escape it.

Regarding the second point, it is still problematic for complex commands. I think its a more general problem that ought to be fixed another way, it seems like we need a foolproof way to do perfect list -> string -> list forwarding in Kakoune. I'd rather keep this terminal change simple and not try to fix it here.

Reply:
Ok I think I see what you're getting at. I've changed the logic of the `terminal` commands so that the argument splitting done by kakoune is preserved. For example, in kakoune if you do
`:terminal printf "%s\n" a b 'c d' 'de''f' &` you will get
```
a
b
c d
de'f
&
```
Consequently, all the escaping logic is moved to the support scripts that need it (i.e. not kitty.kak and not tmux.kak). Anything else that goes through `sh -c` has some logic to preserve the arguments splitting. Some x11 terminals don't need it but go through it anyway so I guess these could still be improved in that regard.

If one actually needs to run things in a shell, they can still do 
`:terminal sh -c 'echo abc; sleep 5'`

If you're happy with this iteration, the iterm script will need some testing as its escaping is rather complicated.

Reply:
Yep, I am happier with that iteration, can you rebase onto master instead of merging it ?